src/pyodide.js

```diff
@@ -619,9 +619,24 @@ globalThis.languagePluginLoader = new Promise((resolve, reject) => {
   };
 
   Module.locateFile = (path) => baseURL + path;
-  Module.postRun = async () => {
-    // Unfortunately the indentation here matters.
-    Module.runPythonSimple(`
+
+  let moduleLoaded = new Promise(r => Module.postRun = r);
+
+  const scriptSrc = `${baseURL}pyodide.asm.js`;
+
+  await loadScript(scriptSrc);
+
+  // The emscripten module needs to be at this location for the core
+  // filesystem to install itself. Once that's complete, it will be replaced
+  // by the call to `makePublicAPI` with a more limited public API.
+  self.pyodide = await pyodide(Module);
```

Comment:
This is not really related to this PR, but I think it would be good to use a different export name:
```sh
-s EXPORT_NAME="'pyodide'"
```
Maybe `make_pyodide_module` or something like that. It's explained that there is a reason that we first put `Module` at `pyodide` and then move it to `pyodide._module`, but there isn't any important reason that we have to reuse the name `pyodide` for three different things?